galaxy.yml

```diff
@@ -0,0 +1,8 @@
+---
+namespace: 'infrawatch'
+name: 'functional_tests'
```

Comment:
I am open to new suggestions, since since these are end-to-end tests, as well as some component functional tests too.
If there's a good suggestion, I would recommend that we also change the name of this repo, to avoid confusion with naming.

Reply:
STF test suite ?
We can split tests into 3 categories. OSP tests is when we test that collectd and metrics_qdr containers are up and running on the OSP overcloud nodes and have metrics locally, then we have STF tests. These are done when we run smoke tests and finally e2e or integration tests when both OSP and STF are involved in the test.  There also will be some STF related tests that will be out of smoke tests scope e.g. prometheus rule creation test that it's purely STF test but not part of smoke tests

Reply:
So, client-side, server-side and end-to-end?

I'd also be inclined to split the tests by scope: single component (incl. regression tests where we check for specific bugs), functional tests for a component, similar to smoketests, but with the actual services generating the data and end to end tests.

I don't think the two lists are mutually exclusive, and we'll need some more granular tests as well.
We can define really granular tests, with a tight scope per role, and then use the playbooks to define test suites 
For example (some hypothetical tests here)
* test-osp-components playbook could include the collectd and qdr roles;
* test-stf would include testing-alerts and testing that the queries we have defined for the dashboards return data; * test end-to-end may include various scenarios including the prom tests we have, but also an autoscaling scenario, if we decide to add that).

TL;DR: we can use roles to define the individual test, and use playbooks to group them.